bigflow/workflow.py

```diff
@@ -45,24 +74,53 @@ def __init__(
         self.start_time_factory = start_time_factory
         self.log_config = log_config
 
-    def run(self, runtime: Optional[str] = None):
-        if runtime is None:
-            runtime = self._auto_runtime()
+    def _parse_runtime_str(self, rt: str):
+        for fmt in self.RUNTIME_FORMATS:
+            try:
+                return dt.datetime.strptime(rt, fmt)
+            except ValueError:
+                pass
+        raise ValueError("Unable to parse 'run_time' %r" % rt)
+
+    def _execute_job(self, job, context):
+        if not isinstance(job, Job):
+            logger.warn("Please, inherit your job %r from `bigflow.Job` class", job)
```

Comment:
Currently, a user has no need to inherit from the interface, so why should he? It's an additional thing to:

* know about
* remember
* do

Reply:
I see two reasons:
 - it might be easier to add further changes/features to 'bigflow' without breaking existing code (so it more like a 'good-practice');
 - better support from ide/editors
    for example in my case i type "def exe<ENTER>" and editor (vscode) complete it to "def execute(self, context: JobContext):" automatically; and I have autocompletion for 'context.' variable.

Reply:
**better support from ide/editors
for example in my case i type "def exe" and editor (vscode) complete it to "def execute(self, context: JobContext):" automatically; and I have autocompletion for 'context.' variable.**

That's a good point. I'm not against that this interface exists, I just don't like the idea of forcing users to use it without a clear reason. If some want to use it for IDE - just do it.

**it might be easier to add further changes/features to 'bigflow' without breaking existing code (so it more like a 'good-practice');**

* if we really want to force users to use this interface, for any reason, then warn is not enough. 
* If we decide to add some features related to the interface, most users still get breaking changes.
* Python has duck-typing, we should depend on the behavior and avoid type checking.
* "we might need it in the future" we don't know, but we know it has a cost right now.

I leave that for @iwko @bartoszwalacik to decide.